thirdeye/thirdeye-pinot/src/main/resources/schema/create-schema.sql

```diff
@@ -437,3 +437,15 @@ create index rootcause_template_id_idx ON rootcause_template_index(base_id);
 create index rootcause_template_owner_idx ON rootcause_template_index(owner);
 create index rootcause_template_metric_idx on rootcause_template_index(metric_id);
 create index rootcause_template_config_application_idx ON rootcause_template_index(`application`);
+
+create table if not exists online_detection_data_index (
+    base_id bigint(20) not null,
+    dataset varchar(200),
+    metric varchar(200),
+    create_time timestamp default 0,
+    update_time timestamp default current_timestamp,
+    version int(10)
+) ENGINE=InnoDB;
```

Comment:
For long term, we might want to consider using BLOB/CLOB to storing ad hoc data so that we can support dataset larger than 64KB, not to mention that the data is stored in JSON format, which is really inefficient ways to store data, compare to any binary format.

Reply:
Thanks for this! I agree with you. In long term, we should use a more efficient way to store the data. This is just current workaround. And if we use a type to store it such as the BLOB, we cannot store it into the `generic_json_entity` table and need to consider where to store it.